src/CodeGenerator/FactoryGenerator.php

```diff
@@ -134,7 +134,7 @@ private function buildParametersCode(string $type)
         $tab = str_repeat(' ', $intention);
         $code = '';
 
-        if (count($withOptions)) {
```

Comment:
This should be imported with `use function` for performance. You may actively do this, but don't worry, I can do this during the pull as well.

Reply:
truthy check is faster than `count($var)`

Reply:
Shouldn't make a difference with the current opcode optimisations in the language, but yes, a direct truth check is faster indeed.

Reply:
Misread of the diff on my side never mind.

Reply:
@Ocramius VLD opcodes for 7.1 show that `count()` actually does function call, initializing function context is the most expensive op here:
```
line     #* E I O op                           fetch          ext  return  operands
-------------------------------------------------------------------------------------
if ($var) :
   5     6      > JMPZ                                                     !0, ->8
   6     7    >   ECHO                                                     1
if (count($var)) :
   9     8    >   INIT_FCALL                                               'count'
         9        SEND_VAR                                                 !0
        10        DO_ICALL                                         $4      
        11      > JMPZ                                                     $4, ->13
  10    12    >   ECHO                                                     2
if (! empty($var)) :
  13    13    >   ISSET_ISEMPTY_VAR                           293601280  ~5      !0
        14        BOOL_NOT                                         ~6      ~5
        15      > JMPZ                                                     ~6, ->17
  14    16    >   ECHO                                                     3
```

Not that it really matters